django/contrib/auth/password_validation.py

```diff
@@ -170,14 +170,22 @@ class CommonPasswordValidator:
     DEFAULT_PASSWORD_LIST_PATH = Path(__file__).resolve().parent / 'common-passwords.txt.gz'
 
     def __init__(self, password_list_path=DEFAULT_PASSWORD_LIST_PATH):
+        # Be careful about de-duplicating the set comprehension with
+        # a try/except/with combination; OSError will only be raised
+        # after iterating through the file, *not* directly from
+        # gzip.open().
```

Comment:
I think this comment is a little bit too explicit for Django's general style. Any changes to the code will probably trigger this case, if there are enough tests. If not - add some! :)

Reply:
Fair.  Well, as we've seen above, tests _will_ already fail if this change is made in the way the comment warns about.